src/include/migraphx/op/softmax.hpp

```diff
@@ -18,10 +18,23 @@ namespace op {
 
 struct softmax
 {
+    int axis = 1;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.axis, "axis"));
+    }
+
     std::string name() const { return "softmax"; }
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs}.has(1).only_dims(4);
+        check_shapes{inputs}.has(1).standard();
+        if(axis < 0 || axis >= inputs[0].lens().size())
+        {
+            MIGRAPHX_THROW("SoftMax: input axis value " + std::to_string(axis) +
```

Comment:
maybe we can add a shape test to trigger an exception for better code coverage.

Reply:
I combined the shape tests for softmax and logsoftmax (with the edge cases on axis accounted for) with the latest commit.

Reply:
Thanks